scripts/github-status.coffee

```diff
@@ -0,0 +1,57 @@
+# Description:
+#   Show current GitHub status and messages
+#
+# Dependencies:
+#   None
+#
+# Configuration:
+#   None
+#
+# Commands:
+#   hubot github status - Returns the current system status and timestamp.
+#   hubot github status last - Returns the last human communication, status, and timestamp.
+#   hubot github status messages - Returns the most recent human communications with status and timestamp.
+#
+# Author:
+#   voke
+
+module.exports = (robot) ->
+  robot.respond /github status$/i, (msg) ->
+    status msg
+
+  robot.respond /github status last$/i, (msg) ->
+    lastMessage msg
+
+  robot.respond /github status messages$/i, (msg) ->
+    statusMessages msg
+
+# NOTE: messages contains new lines for some reason.
+formatString = (string) ->
+  decodeURIComponent(string.replace(/(\n)/gm," "))
+
+status = (msg) ->
+  msg.http('https://status.github.com/api/status.json')
+    .get() (err, res, body) ->
+      json = JSON.parse(body)
+      now = new Date()
+      date = new Date(json['last_updated'])
+      secondsAgo = Math.round((now.getTime() - date.getTime()) / 1000)
+      msg.send "Status: #{json['status']} (#{secondsAgo} seconds ago)"
+
+lastMessage = (msg) ->
+  msg.http('https://status.github.com/api/last-message.json')
```

Comment:
This is now deprecated in favor of https://www.githubstatus.com which has a different API

Reply:
Confirmed

```
$ curl -I  https://status.github.com/api/last-message.json
HTTP/2 301
server: awselb/2.0
date: Tue, 28 May 2019 14:08:17 GMT
content-type: text/html
content-length: 150
location: https://www.githubstatus.com:443/
```

One of the new endpoints: https://www.githubstatus.com/api#status.

```
$ curl -s https://kctbh9vrtdwd.statuspage.io/api/v2/status.json | jq
{
  "page": {
    "id": "kctbh9vrtdwd",
    "name": "GitHub",
    "url": "https://www.githubstatus.com",
    "time_zone": "Etc/UTC",
    "updated_at": "2019-05-28T08:07:23.123Z"
  },
  "status": {
    "indicator": "none",
    "description": "All Systems Operational"
  }
}
```

Apparently they're using Atlassian's statuspage.io. I don't know what's happening. I'm going to go do something else now.

Reply:
`scripts/github-status.coffee` → 🗑 https://github.com/TheLevelUp/hubot-ralph/pull/21/commits/19ee5b8ef51057d4923de55c818da3e8c64d8d8b